applications/matter_weather_station/README.rst

```diff
@@ -156,6 +121,52 @@ Configuration
 
 |config|
 
+.. _matter_weather_station_app_build_types:
+
+Matter weather station build types
+==================================
+
+The Matter weather station application does not use a single :file:`prj.conf` file.
+Configuration files are provided for different build types and they are located in the :file:`configuration/thingy53_nrf5340_cpuapp` directory.
+
+The :file:`prj.conf` file represents a ``debug`` build type.
+Other build types are covered by dedicated files with the build type added as a suffix to the ``prj`` part, as per the following list.
+For example, the ``release`` build type file name is :file:`prj_release.conf`.
+If a board has other configuration files, for example associated with partition layout or child image configuration, these follow the same pattern.
+
+.. include:: /config_and_build/modifying.rst
+   :start-after: build_types_overview_start
+   :end-before: build_types_overview_end
+
+Before you start testing the application, you can select one of the build types supported by Matter weather station application, depending on the building method.
+This application supports the following build types:
+
+* ``debug`` - Debug version of the application.
+  You can use this version to enable additional features for verifying the application behavior, such as logs or command-line shell.
+* ``release`` - Release version of the application.
+  You can use this version to enable only the necessary application functionalities to optimize its performance.
+
+.. note::
+    Currently, this application supports only the ``release`` build type when `Building for the nRF7002 Wi-Fi expansion board`_.
+
+.. note::
+    `Selecting a build type`_ is optional.
+    The ``debug`` build type is used by default if no build type is explicitly selected.
+
+.. _matter_weather_station_app_build_configuration_overlays:
+
+Matter weather station build configuration overlays
+===================================================
+
+* ``overlay-factory_data`` - factory data storage support enabled.
+  You can use this optional configuration overlay to enable reading necessary factory data from a separate partition in the device non-volatile memory.
+  This way, you can read information such as product information, keys, and certificates, useful for example for Matter certification.
+  See `Generating factory data`_ to learn how to put factory data into device's storage.
+  To learn more about factory data, read the :doc:`matter:nrfconnect_factory_data_configuration` page in the Matter documentation.
+
+.. note::
+   Matter factory data support requires the dedicated partition layout, therefore if you build the application using ``overlay-factory_data``, it will not be compatible with other :ref:`Thingy:53 applications and samples <thingy53_compatible_applications>`.
```

Comment:
```suggestion
   Matter factory data support requires the dedicated partition layout.
   This means that if you build the application using the ``overlay-factory_data`` configuration overlay, it will not be compatible with other :ref:`Thingy:53 applications and samples <thingy53_compatible_applications>`.
```